slatedb/src/merge_operator.rs

```diff
@@ -517,40 +629,172 @@ mod tests {
                 }
             }
         }
+
+        // Override merge_batch to handle batches efficiently
+        // Note: merge_batch doesn't have access to the key, so we can't route based on prefix.
+        // For this test, we'll use the default implementation which falls back to pairwise merge()
```

Comment:
let's add the key into the interface (#980 was merged between when you started this PR but we should definitely support it in `merge_batch`)